packages/perseus-core/src/parse-perseus-json/perseus-parsers/widgets-map.test.ts

```diff
@@ -75,16 +74,27 @@ describe("parseWidgetsMap", () => {
         );
     });
 
-    it("rejects an unknown widget type", () => {
+    it("accepts an unknown widget type", () => {
+        // The widget may have been dynamically registered, so we allow it.
         const widgetsMap: unknown = {
             "transmogrifier 1": {type: "transmogrifier"},
         };
 
         const result = parse(widgetsMap, parseWidgetsMap);
 
+        expect(result).toEqual(success(widgetsMap));
+    });
+
+    it("rejects an unknown widget whose type doesn't match its ID", () => {
+        const widgetsMap: unknown = {
+            "transmogrifier 1": {type: "bork"},
```

Comment:
I'm not sure how much we want a test like this. We try to keep our understanding of widget IDs to be opaque blobs with the ultimate goal of having arbitrary IDs in the future (`widget 1` >> `895h-789e-39j2-4359`). In practice we don't do this, but I don't think we want to move towards reinforcing the idea that IDs contain widget types.